### ISSUE TITLE
MLModelDeploymentsUpgradeIT full-bwc tests failing with `processor [pipeline] doesn't support one or more provided configuration parameters [ignore_missing_pipeline]`

### ISSUE THREAD

**Issue by nik9000** [elastic/elasticsearch#108976]
### CI Link

none

### Repro line

./gradlew ':x-pack:qa:rolling-upgrade:v8.1.3#twoThirdsUpgradedTest' -Dtests.class="org.elasticsearch.upgrades.MLModelDeploymentsUpgradeIT" -Dtests.method="testTrainedModelDeployment" -Dtests.seed=FC3B029B17FA911D -Dtests.bwc=true -Dtests.locale=fr-BE -Dtests.timezone=Africa/Sao_Tome -Druntime.java=22

### Does it reproduce?

Yes

### Applicable branches

main

### Failure excerpt

https://gradle-enterprise.elastic.co/s/cslmp2wcxnnc4/tests/task/:x-pack:qa:rolling-upgrade:v8.1.3%23twoThirdsUpgradedTest/details/org.elasticsearch.upgrades.MLModelDeploymentsUpgradeIT/testTrainedModelDeployment?top-execution=1
```
{"error":{"root_cause":[{"type":"parse_exception","reason":"processor [pipeline] doesn't support one or more provided configuration parameters [ignore_missing_pipeline]","processor_type":"pipeline","suppressed":[{"type":"parse_exception","reason":"processor [pipeline] doesn't support one or more provided configuration parameters [ignore_missing_pipeline]","processor_type":"pipeline"},{"type":"parse_exception","reason":"processor [pipeline] doesn't support one or more provided configuration parameters [ignore_missing_pipeline]","processor_type":"pipeline"}]}],"type":"exception","reason":"unexpected failure gathering pipeline information","caused_by":{"type":"parse_exception","reason":"processor [pipeline] doesn't support one or more provided configuration parameters [ignore_missing_pipeline]","processor_type":"pipeline","suppressed":[{"type":"parse_exception","reason":"processor [pipeline] doesn't support one or more provided configuration parameters [ignore_missing_pipeline]","processor_type":"pipeline"},{"type":"parse_exception","reason":"processor [pipeline] doesn't support one or more provided configuration parameters [ignore_missing_pipeline]","processor_type":"pipeline"}]}},"status":500}
```

**Comment by andreidan** [comment]
Caused by https://github.com/elastic/elasticsearch/pull/87354 introduced in `8.4`

**Comment by maxhniebergall** [comment]
Given that this seems to be a serialization error that was introduced in 8.4, this seems like a won't fix since we don't support 8.4 anymore.
